### PR TITLE
[PM-35120] - Update user delete flows with org ownership

### PR DESCRIPTION
**Pull request by jaasen-livefront**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-35120

## 📔 Objective

Fixes a silent data-loss bug where a user who is the sole owner of an organization could delete their account without any warning, which would also silently delete the organization(s) they own.

The server already blocks deletion for paid-plan org owners (Enterprise, Families, Teams, etc.) with a generic error, but free-org owners had no protection at all.

**New behaviour:**

- **Paid-org owner** – a blocking error dialog is shown explaining they must transfer ownership or delete the organization before deleting their account. The flow stops here (avoids the confusing server-side error).
- **Free-org owner** – a warning dialog is shown explaining that deleting their account will also delete any organizations they are the sole owner of. The user can cancel or confirm and proceed.
- **No org / not an owner** – existing flow is unchanged; the Delete Account dialog opens directly.

Only confirmed (`Confirmed` status) ownership memberships are considered.

## 🏗️ Changes

- Introduced `AccountDeletionService` (`libs/angular`) — encapsulates the full account-deletion guard/flow logic (org ownership checks, warning/blocking dialogs, and opening `DeleteAccountDialogComponent`). Replaces the previous one-liner call at each call site.
- Wired `AccountDeletionService` into the desktop `AppComponent` and the web `AccountComponent`.
- Added missing `deleteAccountOrganizationOwnerWarning` and `cannotDeleteAccountOrganizationOwnerDesc` translation keys to the desktop `en` and `en_IN` locales.

### Web
https://github.com/user-attachments/assets/d2fcdfd3-af35-46b3-ae51-59fe25341a99

### Desktop
https://github.com/user-attachments/assets/c3d6e3f9-8f20-42f6-b104-8b20fe63c907

### CODE REVIEW

File: apps/desktop/src/locales/en_IN/messages.json [hunk 1]
```diff
@@ -2161,6 +2161,12 @@
   "deleteAccountWarning": {
     "message": "Deleting your account is permanent. It cannot be undone."
   },
+  "deleteAccountOrganizationOwnerWarning": {
+    "message": "You are the owner of at least one organisation. Deleting your account will also delete any organisations you are the sole owner of."
+  },
+  "cannotDeleteAccountOrganizationOwnerDesc": {
+    "message": "You cannot delete your account while you are the owner of an organisation. Transfer ownership or delete the organisation first."
+  },
```
Comment (djsmith85) [inline apps/desktop/src/locales/en_IN/messages.json]: ⚠️ There should be no addition to this file. Please revert. Changes are synced via Crowdin Sync PR

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/desktop/src/locales/en_IN/messages.json (hunk 1 above)
Comment (jaasen-livefront) [reply apps/desktop/src/locales/en_IN/messages.json]: @djsmith85 Ack! Fixed. https://github.com/bitwarden/clients/pull/20462/commits/c0c3f14754124450342227a54b9c320c511613db

**Review by jengstrom-bw — DISMISSED** [review]
Looks good!

**Review by djsmith85 — DISMISSED** [review]
Thank you for removing the changes to `apps/desktop/src/locales/en_IN/messages.json`

File: libs/angular/src/auth/account-deletion/account-deletion.service.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  async openDeleteAccountFlow(): Promise<void> {
+    const userId = await firstValueFrom(getUserId(this.accountService.activeAccount$));
+    const organizations = await firstValueFrom(this.organizationService.organizations$(userId));
+
+    const userIsManaged = organizations.some((o) => o.userIsManagedByOrganization === true);
+
+    if (userIsManaged) {
+      await this.dialogService.openSimpleDialog({
+        title: { key: "cannotDeleteAccount" },
+        content: { key: "cannotDeleteAccountDesc" },
```
Comment (JaredSnider-Bitwarden) [inline libs/angular/src/auth/account-deletion/account-deletion.service.ts:30]: ⚠️ :  These i18n keys aren't defined in the web's translations file.

File: libs/angular/src/auth/account-deletion/account-deletion.service.ts [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+      (org) => org.isOwner && org.isMember && org.status === OrganizationUserStatusType.Confirmed,
+    );
+
+    const ownsPaidOrg = confirmedOwnerOrgs.some(
+      (org) => org.productTierType !== ProductTierType.Free,
+    );
+
+    if (ownsPaidOrg) {
+      await this.dialogService.openSimpleDialog({
+        title: { key: "deleteAccount" },
```
Comment (JaredSnider-Bitwarden) [inline libs/angular/src/auth/account-deletion/account-deletion.service.ts]: 🎨 ❓ :  Should this be `cannotDeleteAccount` since the user can't actually delete their account b/c they are the org owner?

File: libs/angular/src/auth/account-deletion/account-deletion.service.spec.ts
Comment (JaredSnider-Bitwarden) [inline libs/angular/src/auth/account-deletion/account-deletion.service.spec.ts:1]: Thank you for writing these tests! 

Do you think it's worth writing tests to cover these cases (claude found these):

**Owner filter edge cases**  — (`org.isOwner && org.isMember && status === Confirmed`):

⚠️ No test for `isMember = false` (provider-only owner). This is the security-relevant case called out in the review summary — currently relying on the filter to exempt providers, but unprotected by a test.
⚠️ No test for owner with `status !== Confirmed` (Invited/Accepted/Revoked) — should skip gating.
Paid-tier coverage:

⚠️ Only `Enterprise` is exercised. `TeamsStarter`, `Teams`, `Families` all go through the same `productTierType !== Free` check, but a parameterized test would future-proof against someone adding a new tier that should bypass gating.

**Mixed-org priority**:

⚠️ No test for "user owns both a free and a paid org" — paid path should win (evaluated first). Currently implicit.

File: libs/angular/src/auth/account-deletion/account-deletion.service.ts
Comment (JaredSnider-Bitwarden) [inline libs/angular/src/auth/account-deletion/account-deletion.service.ts:1]: Thank you for making a feature folder here! 

❓ : can you move the `libs/angular/src/auth/delete-account-dialog` folder under this one to consolidate the related files?

**Review by JaredSnider-Bitwarden — CHANGES_REQUESTED** [review]
This is shaping up nicely. I just have a few items below 👇

File: libs/angular/src/auth/account-deletion/account-deletion.service.ts (hunk 2 above)
Comment (jaasen-livefront) [reply libs/angular/src/auth/account-deletion/account-deletion.service.ts:30]: Fixed https://github.com/bitwarden/clients/pull/20462/commits/9629360b0df659fd4a05677ccdb47348b23be8c0

File: libs/angular/src/auth/account-deletion/account-deletion.service.ts (hunk 3 above)
Comment (jaasen-livefront) [reply libs/angular/src/auth/account-deletion/account-deletion.service.ts]: Agreed. Updated https://github.com/bitwarden/clients/pull/20462/commits/9629360b0df659fd4a05677ccdb47348b23be8c0

File: libs/angular/src/auth/account-deletion/account-deletion.service.spec.ts
Comment (jaasen-livefront) [reply libs/angular/src/auth/account-deletion/account-deletion.service.spec.ts:1]: Agree these are nice casese to catch.

https://github.com/bitwarden/clients/pull/20462/commits/9629360b0df659fd4a05677ccdb47348b23be8c0

File: libs/angular/src/auth/account-deletion/account-deletion.service.ts
Comment (jaasen-livefront) [reply libs/angular/src/auth/account-deletion/account-deletion.service.ts:1]: Done! https://github.com/bitwarden/clients/pull/20462/commits/9629360b0df659fd4a05677ccdb47348b23be8c0

**Review by JaredSnider-Bitwarden — DISMISSED** [review]
LGTM! Thank you!

**Review by jengstrom-bw — DISMISSED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by gbubemismith — DISMISSED** [review]

**Comment by jaasen-livefront** [comment]
@gbubemismith @JaredSnider-Bitwarden @jengstrom-bw  @djsmith85 For some reason GH dismissed your reviews after merging main. Would you mind re-approving? Ty!

**Review by jengstrom-bw — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]
